src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -91,6 +91,9 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
     getParameterValues: (id) => BINDING.js_string_to_mono_string(componentAttacher.getParameterValues(id) || ''),
   };
 
+  const persistedState = discoverPersistedState(document);
+  window['Blazor']._internal.getPersistedState = () => BINDING.js_string_to_mono_string(persistedState);
```

Comment:
Could we defer the scan through the document until the .NET side asks for it? Otherwise everyone has to pay the cost even if they aren't using the feature. I know this is not a massively expensive process but sooner or later we're going to have to streamline the startup process as much as we can, so would be good to avoid adding more things to it if it's simple to avoid them.

```suggestion
  window['Blazor']._internal.getPersistedState = () => BINDING.js_string_to_mono_string(discoverPersistedState(document));
```